core/store/src/trie/mod.rs

```diff
@@ -416,6 +416,60 @@ pub struct Trie {
     pub counter: TouchedNodesCounter,
 }
 
+pub struct ShardTries {
+    tries: Vec<Arc<Trie>>,
```

Comment:
I'm surprised that we use it outside in such way: `Arc<Vec<Arc<Trie>>>`. I expect it should be just `Vec<Arc<Trie>>`. What is the reason to keep Arc for Vec?

Reply:
It seems to me that the outer one is the one that is potentially shared between threads. So shouldn't it be `Arc<Vec<Trie>>`?

Reply:
`Arc<Trie>` to pass to `TrieUpdate`
`Arc<Vec<Arc<Trie>>>` to pass the whole thing all around without copying the Vec
Later we can refactor the code to non-owning references and lifetime annotations, Arc just makes things simpler

Reply:
@bowenwang1996 it doesn't make sense because different threads may have different vectors filled with same `Arc<Trie>` data. No need to do `Arc<Vec<Arc<..>>>`.

@mikhailOK why not copy a vector? It's simple and transparent.

Reply:
Having one instance of `ShardTries` is easier to reason about than having copies.

Here is a new issue for a refactor: #2639, this is potentially not the `Arc` to remove.

Reply:
Sounds good.